src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -0,0 +1,1242 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.ServiceModel.Dispatcher;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Extensions;
+using System.Xml.Schema;
+using static System.Xml.Serialization.XmlSerializationReaderILGen;
+
+namespace System.Xml.Serialization
+{
+    internal class ReflectionXmlSerializationReader : XmlSerializationReader
+    {
+        private static TypeDesc StringTypeDesc { get; set; } = (new TypeScope()).GetTypeDesc(typeof(string));
+        private static TypeDesc QnameTypeDesc { get; set; } = (new TypeScope()).GetTypeDesc(typeof(XmlQualifiedName));
+
+        private XmlMapping _mapping;
+
+        public ReflectionXmlSerializationReader(XmlMapping mapping, XmlReader xmlReader, object events, string encodingStyle)
+        {
+            Init(xmlReader, events, encodingStyle, tempAssembly: null);
+            _mapping = mapping;
+        }
+
+        protected override void InitCallbacks()
+        {
+        }
+
+        protected override void InitIDs()
+        {
+        }
+
+        internal object ReadObject()
+        {
+            var xmlMapping = _mapping;
+            if (!xmlMapping.IsReadable)
+                return null;
+            if (!xmlMapping.GenerateSerializer)
+                throw new ArgumentException(SR.Format(SR.XmlInternalError, "xmlMapping"));
+
+            if (xmlMapping is XmlTypeMapping)
+                return GenerateTypeElement((XmlTypeMapping)xmlMapping);
+            else if (xmlMapping is XmlMembersMapping)
+                return GenerateMembersElement((XmlMembersMapping)xmlMapping);
+            else
+                throw new ArgumentException(SR.Format(SR.XmlInternalError, "xmlMapping"));
+        }
+
+        private object GenerateMembersElement(XmlMembersMapping xmlMapping)
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
>throw new PlatformNotSupportedException(); [](start = 12, length = 42)

Could you add a comment explaining why this throws PNSE?  Is this an issue we intend to fix, or is it by-design?